### PR TITLE
Add letter validation to hangman

### PR DESCRIPTION
**Pull request by mattstates**
I wasn't able to build it locally, but I think this will work for you.

### CODE REVIEW

File: src/DevChatter.Bot.Core/Games/Hangman/HangmanGame.cs [hunk 1]
```diff
@@ -130,12 +130,16 @@ public void AskAboutLetter(IChatClient chatClient, string letterToAsk, ChatUser
...
                 chatClient.SendMessage($"Yep, {letterToAsk} is in here. {MaskedPassword}");
             }
-            else
+            else if (ALL_LETTERS.Contains(letterToAsk))
             {
                 chatClient.SendMessage($"No, {letterToAsk} is not in the word.");
                 _hangmanDisplayNotification.HangmanWrongAnswer();
                 CheckForGameLost(chatClient);
             }
+            else
```
Comment (benrick) [inline src/DevChatter.Bot.Core/Games/Hangman/HangmanGame.cs]: I like it!

Perhaps move this before the `_guessedLetters.Add(new HangmanGuess(letterToAsk, chatUser));` line and return, so we don't add this invalid choice to the collection. That collection is how we're determining a lost game.

Comment (mattstates) [reply src/DevChatter.Bot.Core/Games/Hangman/HangmanGame.cs]: Good catch, I will fix this later today.

**Comment by mattstates** [comment]
I moved the validation before the letter is added to the guessedLetter list.

**Comment by mattstates** [comment]
Thanks benrick.

**Pull request merged**